src/main/java/com/connexta/poller/service/TransformPollingService.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.poller.service;
+
+import com.connexta.poller.api.PollingService;
+import com.connexta.poller.api.PollingTask;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+/**
+ * This is class is meant to be instantiated as a Bean, Component, or Service. The class's
+ * responsibilities are to manage the settings for polling and kick off new polling tasks. Other
+ * responsibilities include settings the wait and retry behavior. It also creates and configures the
+ * executor, and ensures that it is shutdown properly.
+ */
+@Service
+@Slf4j
+public class TransformPollingService implements PollingService {
```

Comment:
❓ Do we really need a polling interfaces? I think this class would do just fine as a bean with an internal timer on when it runs. I would imagine that we would not expect any extending of our system's capabilities through these interfaces.

Reply:
I don't know. I've been dinged in the past for not following Java best practices.  Java best practices includes lots of interfaces. In the long run I don't think using the interfaces or not using the interfaces is going to  make a big difference. Other people may have stronger opinions on the matter and I'm totally open to that. I wish GitHub had surveys for this kind of thing : "(A) The interfaces stay. (B) The interfaces go."